src/bitmap/store.rs

```diff
@@ -43,6 +43,69 @@ impl Store {
         }
     }
 
+    pub fn insert_range(&mut self, range: Range<u16>) -> u64 {
+        // A Range is defined as being of size 0 if start >= end.
+        if range.is_empty() {
+            return 0;
+        }
+
+        match *self {
+            Array(ref mut vec) => {
+                let (range_start, range_end) = (range.start, range.end);
```

Comment:
Was wondering why you are doing that? maybe du to some previous work? :)

Reply:
`Range<Idx>` is a bit weird - the inner `Idx` are not copy and therefore neither is the `Range` so it _moves_ into the `vec.splice(..., range)` call.

This consumes the range, making the "end - start - dropped" maths at the end impossible. I've pushed a change that clones the range for the splice call though and it is indeed a bit cleaner  👍